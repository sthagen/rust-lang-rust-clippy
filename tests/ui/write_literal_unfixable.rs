//@no-rustfix
#![allow(unused_must_use)]
#![warn(clippy::write_literal)]

use std::io::Write;

fn escaping() {
    let mut v = vec![];

    writeln!(v, r"{}", '"');
    //~^ write_literal

    // hard mode
    writeln!(v, r#"{}{}"#, '#', '"');
    //~^ write_literal
}
