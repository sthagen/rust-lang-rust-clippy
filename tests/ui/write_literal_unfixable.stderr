error: literal with an empty format string
  --> tests/ui/write_literal_unfixable.rs:10:24
   |
LL |     writeln!(v, r"{}", '"');
   |                        ^^^
   |
   = note: `-D clippy::write-literal` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::write_literal)]`

error: literal with an empty format string
  --> tests/ui/write_literal_unfixable.rs:14:28
   |
LL |     writeln!(v, r#"{}{}"#, '#', '"');
   |                            ^^^^^^^^

error: aborting due to 2 previous errors

